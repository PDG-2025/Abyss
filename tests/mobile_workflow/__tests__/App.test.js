import React from 'react';
import {render, screen} from '@testing-library/react-native';
import App from '../App';

// Mock pour useColorScheme
jest.mock('react-native/Libraries/Utilities/useColorScheme', () => ({
  __esModule: true,
  default: jest.fn(() => 'light'),
}));

describe('App Component', () => {
  test('renders Hello World text', () => {
    render(<App />);
    
    const welcomeText = screen.getByTestId('welcome-text');
    expect(welcomeText).toBeTruthy();
    expect(welcomeText.props.children).toBe('Hello World!');
  });

  test('renders welcome subtitle', () => {
    render(<App />);
    
    const subtitleText = screen.getByTestId('subtitle-text');
    expect(subtitleText).toBeTruthy();
    expect(subtitleText.props.children).toBe('Welcome to React Native');
  });

  test('renders info text about the app', () => {
    render(<App />);
    
    const infoTexts = screen.getAllByText(/This is a simple React Native app/i);
    expect(infoTexts.length).toBeGreaterThan(0);
  });

  test('renders CI/CD info text', () => {
    render(<App />);
    
    const cicdText = screen.getByText(/Built for CI\/CD with GitHub Actions/i);
    expect(cicdText).toBeTruthy();
  });

  test('app component matches snapshot', () => {
    const tree = render(<App />);
    expect(tree).toMatchSnapshot();
  });
});
