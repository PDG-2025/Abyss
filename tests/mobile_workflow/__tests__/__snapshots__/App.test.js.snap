// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component app component matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "minHeight": 600,
            "padding": 20,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 32,
                "fontWeight": "700",
                "marginBottom": 16,
                "textAlign": "center",
              },
              {
                "color": "#000000",
              },
            ]
          }
          testID="welcome-text"
        >
          Hello World!
        </Text>
        <Text
          style={
            [
              {
                "fontSize": 24,
                "fontWeight": "400",
                "marginBottom": 32,
                "textAlign": "center",
              },
              {
                "color": "#000000",
              },
            ]
          }
          testID="subtitle-text"
        >
          Welcome to React Native
        </Text>
        <View
          style={
            {
              "marginTop": 40,
              "paddingHorizontal": 20,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginBottom": 12,
                  "textAlign": "center",
                },
                {
                  "color": "#000000",
                },
              ]
            }
          >
            This is a simple React Native app with automated testing.
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 16,
                  "lineHeight": 24,
                  "marginBottom": 12,
                  "textAlign": "center",
                },
                {
                  "color": "#000000",
                },
              ]
            }
          >
            Built for CI/CD with GitHub Actions.
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
